src/fsharp/TastOps.fs

```diff
@@ -5863,16 +5863,20 @@ let rec mkExprAddrOfExprAux g mustTakeAddress useReadonlyForGenericArrayAddress
             let ty = tyOfExpr g expr
             if isStructTy g ty then 
                 match mut with 
-                | NeverMutates -> ()
-                | AddressOfOp -> 
-                    // we get an inref
-                    errorR(Error(FSComp.SR.tastCantTakeAddressOfExpression(), m))
                 | DefinitelyMutates -> 
                     // Give a nice error message for mutating something we can't take the address of
                     errorR(Error(FSComp.SR.tastInvalidMutationOfConstant(), m))
                 | PossiblyMutates -> 
                     // Warn on defensive copy of something we can't take the address of
                     warning(DefensiveCopyWarning(FSComp.SR.tastValueHasBeenCopied(), m))
+                | _ -> ()
```

Comment:
Code cleanup: please don't use a catch-all pattern here, list out the cases explicitly in delicate code.

Reply:
Will do. Thanks.